tee-worker/litentry/core/identity-verification/src/web3/mod.rs

```diff
@@ -112,25 +121,27 @@ fn verify_substrate_signature_internal(
 fn verify_evm_signature(
 	who: &AccountId,
 	identity: &Identity,
-	code: &ChallengeCode,
+	sidechain_nonce: Index,
+	key: UserShieldingKeyType,
+	nonce: UserShieldingKeyNonceType,
 	validation_data: &Web3CommonValidationData,
 ) -> Result<()> {
-	let msg = get_expected_raw_message(who, identity, code);
+	let msg = get_expected_raw_message(who, identity, sidechain_nonce, key, nonce);
 	let digest = compute_evm_msg_digest(&msg);
 	let evm_address = if let Identity::Evm { address, .. } = identity {
 		address
 	} else {
-		return Err(Error::VerifyIdentityFailed(ErrorDetail::InvalidIdentity))
+		return Err(Error::LinkIdentityFailed(ErrorDetail::InvalidIdentity))
```

Comment:
Same here, I think it's unnecessarily coupled to identity linking usecase.

Reply:
I'll be against this though. The error will be bubbled to F/E in the end.

There's only one `link_identity` ext now, so you either expect `IdentityLinked` or `LinkIdentityFailed`. Having additional error events might be confusing.
What do you think?

Reply:
I though the error could be mapped, but yes on the other hand it would make the code more complex.
I'm still in favor of getting rid of all `link` mentions here if it's possible but we can keep it as it is as well if you are not convinced enough 👍🏼